src/views/default-view.ts

```diff
@@ -385,6 +386,49 @@ export class DefaultView<N extends INodeBase, E extends IEdgeBase> implements IO
     }
   };
 
+  mouseRightClicked = (event: PointerEvent) => {
+    // event.preventDefault();
+    const mousePoint = this.getCanvasMousePosition(event);
+    const simulationPoint = this._renderer.getSimulationPosition(mousePoint);
+
+    if (this._strategy.onMouseRightClick) {
+      const response = this._strategy.onMouseRightClick(this._graph, simulationPoint);
+      const subject = response.changedSubject;
+
+      
+      if (subject) {
+        if (isNode(subject)) {
+          this._events.emit(OrbEventType.NODE_RIGHT_CLICK, {
+            node: subject,
+            event,
+            localPoint: simulationPoint,
+            globalPoint: mousePoint,
+          });
+        }
+        if (isEdge(subject)) {
+          this._events.emit(OrbEventType.EDGE_RIGHT_CLICK, {
+            edge: subject,
+            event,
+            localPoint: simulationPoint,
+            globalPoint: mousePoint,
+          });
+        }
+      } else {
+        this._events.emit(OrbEventType.CANVAS_RIGHT_CLICK, {
+          event,
+          localPoint: simulationPoint,
+          globalPoint: mousePoint,
+        });
+      }
```

Comment:
```suggestion
      }
      
      this._events.emit(OrbEventType.MOUSE_RIGHT_CLICK, {
        subject,
        event,
        localPoint: simulationPoint,
        globalPoint: mousePoint,
      });
```

Just as my comment before regarding the more generic approach with the mouse click. In this case, the subject will be `undefined` where someone using the API can easily know if the mouse right click was on canvas or on top of the node or edge, e.g.

```typescript
orb.events.on(Orb.OrbEventType.MOUSE_RIGHT_CLICK, (event) => {
  if (!event.subject) {
    // This is a canvas click
  }
});
```